### PR TITLE
67 link communicator ecs transisthor inside server client

### PR DESCRIPTION
**Pull request by LucasTesnier**
Implement ClientRoom function.

Add a basic protocol connection between client and server.

It work !

Part 2/2 of the issue.

### CODE REVIEW

File: Client/ClientRoom.hpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        enum ClientState { UNDEFINED, LOBBY, IN_GAME, ENDED };
+
+        /// @brief Construct a new ClientRoom with default value
+        ClientRoom();
+
+        /// @brief Construct a new ClientRoom
+        /// @param address The room address
+        /// @param port room port
+        /// @param serverAddress server address
+        /// @param serverPort room port
```
Comment (31Nathan) [inline Client/ClientRoom.hpp:38]: Room port on server port ?

**Review by 31Nathan — APPROVED** [review]
Only one error one doc, very good job otherwise !!

File: Client/ClientRoom.cpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    _serverEndpoint = Client();
+    _communicatorInstance = std::make_shared<Communicator>(_networkInformations);
+    _worldInstance = std::make_shared<World>(1);
+    _transisthorInstance = std::make_shared<Transisthor>(*(_communicatorInstance.get()), *(_worldInstance.get()));
+    _communicatorInstance.get()->updateTransisthorBridge(_transisthorInstance);
+    _worldInstance.get()->updateTransisthorBridge(_communicatorInstance.get()->getTransisthorBridge());
+    _state = ClientState::UNDEFINED;
+}
+
+ClientRoom::ClientRoom(std::string address, unsigned short port, std::string serverAddress, unsigned short serverPort)
```
Comment (ZQUEMA) [inline Client/ClientRoom.cpp:29]: May be change with `const std:: string &address` and `const std:: string &serverAddress`

File: Client/ClientRoom.hpp [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        ClientRoom(std::string address, unsigned short port, std::string serverAddress, unsigned short serverPort);
```
Comment (ZQUEMA) [inline Client/ClientRoom.hpp:39]: May be change with `const std:: string &address` and const `std:: string &serverAddress`

**Review by ZQUEMA — APPROVED** [review]

File: Client/ClientRoom.hpp (hunk 1 above)
Comment (LucasTesnier) [reply Client/ClientRoom.hpp:38]: Thanks for reporting, it wiil change in the next PR

**Pull request merged**